frontend/src/hooks/dashboard-preview-hooks.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlusSquare, faMinusSquare } from "@fortawesome/free-solid-svg-icons";
+import Button from "../components/Button";
+
+export function useFullPreview() {
+  const [fullPreview, setFullPreview] = useState(false);
+
+  const fullPreviewToggle = () => {
+    setFullPreview(!fullPreview);
+  };
+
+  const fullPreviewButton = fullPreview ? (
+    <Button onClick={fullPreviewToggle} variant="outline" type="button">
+      <FontAwesomeIcon icon={faMinusSquare} className={"margin-right-1"} />
+      Close preview
+    </Button>
+  ) : (
+    <Button onClick={fullPreviewToggle} variant="outline" type="button">
+      <FontAwesomeIcon icon={faPlusSquare} className={"margin-right-1"} />
+      Expand preview
+    </Button>
+  );
+
+  return { fullPreviewToggle, fullPreviewButton, fullPreview };
+}
```

Comment:
Interesting implementation, a hook that returns HTML 🤔 . Never seen this before. I'm pretty sure that hooks were not meant to return components. But I can't think of why this would be a bad idea at the moment either. 

Did you run into performance issues by any chance? Any lagginess in the UI? I'm thinking of a situation where the hook is always returning a new HTML value and thus causing extra renders.

Reply:
Nope, I didn't run into any performance issues.

Reply:
Ok, let's leave it then and keep an eye on it :)